architecture-decisions/0003-remove-memory-tuning.md

```diff
@@ -0,0 +1,39 @@
+# 3. Remove memory tuning
+
+Date: 2018-11-14
+
+## Status
+
+Accepted
+
+## Context
+
+We have seen a number of OutOfMemoryErrors with Solr, mostly related to garbage collection.  Our memory
+settings include many overrides of the defaults, and it's not clear what they do individually or as a group.
+These settings have also been in use for several years, and we're not sure if they are still a good match for
+our hardware and usage.
+
+## Decision
+
+Remove the following memory tuning options from the Java options:
+* `-Xss=256k`
+* `-XX:NewRatio=3`
+* `-XX:SurvivorRatio=4`
+* `-XX:TargetSurvivorRatio=90`
+* `-XX:MaxTenuringThreshold=8`
+* `-XX:ConcGCThreads=4`
+* `-XX:ParallelGCThreads=4`
+* `-XX:PretenureSizeThreshold=64m`
+* `-XX:CMSInitiatingOccupancyFraction=50`
+* `-XX:CMSMaxAbortablePrecleanTime=6000`
+
+Add the following option to disable throwing an OutOfMemoryError if garbage collection takes too long:
+* `-XX:-UseGCOverheadLimit`
+
+
+## Consequences
+
+* Not throwing an OutOfMemoryError may result in Java hanging instead of throwing an error, which would make
+  the problem instead of notifying us.
```

Comment:
can you clarify this first consequence? I am having trouble parsing it grammatically. would make the problem what?

Reply:
Sorry, I meant "hide the problem" instead of "make the problem"

Reply:
ahh that makes more sense! thanks